eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -297,15 +297,23 @@ else if (inside.size() > 32 && !(inside instanceof SetIterable))
      *
      * @since 1.0
      */
-    boolean containsAllIterable(Iterable<?> source);
+    default boolean containsAllIterable(Iterable<?> source)
+    {
+        RichIterable<?> outside = this.toSet();
+        return StreamSupport.stream(source.spliterator(), false).allMatch(outside::contains);
+    }
 
     /**
      * Returns true if all elements in source are contained in this collection.
      *
      * @see Collection#containsAll(Collection)
      * @since 1.0
      */
-    boolean containsAll(Collection<?> source);
+    default boolean containsAll(Collection<?> source)
+    {
+        return source instanceof RichIterable ? this.containsAllIterable(source)
+                : source.stream().allMatch(this::contains);
+    }
```

Comment:
I don't understand the conditional here. What does it matter if `source` is a `RichIterable`? It's being passed to a method that takes `Iterable`.

In addition, containsAllIterable() may convert `this` to a set, but if the collection is a plain one, not a RichIterable, then we don't convert anything to a set. Why the difference? If it's for performance, this inconsistency doesn't make sense to me.